app/views/landing_page/blocks/_main_navigation.html.erb

```diff
@@ -1,45 +1,46 @@
-      <% contents_list(request.path, block.data["links"]).each do |link| %>
-        <%
-          li_classes = "app-b-main-nav__listitem"
-          li_aria = { current: true } if link[:active]
-        %>
-        <%= tag.li(class: li_classes, aria: li_aria) do %>
+    <nav id="app-b-main-nav__nav">
+      <ul class="app-b-main-nav__ul">
+        <% contents_list(request.path, block.links).each do |link| %>
           <%
-            link_classes = "app-b-main-nav__link govuk-link govuk-link--no-underline"
-            link_classes << " app-b-main-nav__link--active" if link[:active]
+            li_classes = "app-b-main-nav__listitem"
+            li_aria = { current: true } if link[:active]
           %>
-          <%= tag.a(href: link[:href], class: link_classes) do %>
-            <span class="app-b-main-nav__mobile-border"></span>
-            <%= link[:text] %>
-          <% end %>
-          <% if link[:items] %>
-            <ul class="app-b-main-nav__childlist">
-              <% link[:items].each do |child_link| %>
-                <%
-                  child_link_classes = "app-b-main-nav__link govuk-link govuk-link--no-underline"
-                  child_link_classes << " app-b-main-nav__link--active" if child_link[:active]
-                %>
-                <li class="app-b-main-nav__listitem">
-                  <%= tag.a(href: child_link[:href], class: child_link_classes) do %>
-                    <span class="app-b-main-nav__mobile-border"></span>
-                    <%= child_link[:text] %>
-                  <% end %>
-                </li>
-              <% end %>
-            </ul>
+          <%= tag.li(class: li_classes, aria: li_aria) do %>
+            <%
+              link_classes = "app-b-main-nav__link govuk-link govuk-link--no-underline"
+              link_classes << " app-b-main-nav__link--active" if link[:active]
+            %>
+            <%= tag.a(href: link[:href], class: link_classes) do %>
+              <span class="app-b-main-nav__mobile-border"></span>
+              <%= link[:text] %>
+            <% end %>
+            <% if link[:items] %>
+              <ul class="app-b-main-nav__childlist">
+                <% link[:items].each do |child_link| %>
+                  <%
+                    child_link_classes = "app-b-main-nav__link govuk-link govuk-link--no-underline"
+                    child_link_classes << " app-b-main-nav__link--active" if child_link[:active]
+                  %>
+                  <li class="app-b-main-nav__listitem">
+                    <%= tag.a(href: child_link[:href], class: child_link_classes) do %>
+                      <span class="app-b-main-nav__mobile-border"></span>
+                      <%= child_link[:text] %>
+                    <% end %>
+                  </li>
+                <% end %>
+              </ul>
+            <% end %>
           <% end %>
         <% end %>
-      <% end %>
-    </ul>
-  </nav>
+      </ul>
+    </nav>
+  </div>
```

Comment:
For ease of review as the diff looks a bit crazy, all I've done in this file is:
- added `<div class="app-b-main-nav--border">` around the block
- added `govuk-width-container` to the first child div
- switched `block.data["title"]` to `block.title`, `block.data["children"] to `block.items`, etc.